stan/math/prim/mat/err/check_matching_dims.hpp

```diff
@@ -47,6 +46,49 @@ namespace stan {
                        "columns of ", name2, y2.cols());
     }
 
+    /**
+     * Check if the two matrices are of the same size.
+     *
+     * This function checks the runtime sizes and can also check the static
+     * sizes as well. For example, a 4x1 matrix is not the same as a vector
+     * with 4 elements.
+     *
+     * @tparam check_compile Whether to check the static sizes 
```

Comment:
When do you not want to check static size match?

Reply:
I ran `grep -r "check_matching_dims" ./` to see where `check_matching_dims` is being used. The first few results were from probability distributions. I can see it not mattering there if the vectors are wrapped in `seq_scalar_view` because the distribution function accesses the vectors' values through `seq_scalar_view`. However, should someone go through and check whether a function using `check_matching_dims` needs the static sizes check?